framework/vst/internal/plugineditorview.cpp

```diff
@@ -0,0 +1,156 @@
+}
+
+PluginEditorView::~PluginEditorView()
+{
+    if (m_activeViews[instanceId()]) {
+        m_activeViews[instanceId()] = nullptr;
+    }
+    if (m_view) {
+        m_view->removed();
+    }
+}
+
+int PluginEditorView::metaTypeId()
+{
+    return QMetaType::type("PluginEditorView");
+}
+
+void PluginEditorView::setInstanceId(IVSTInstanceRegister::instance_id id)
+{
+    IF_ASSERT_FAILED(m_instanceId == NOT_SETTED) {
+        LOGE() << "can't reset instance id";
+        return;
+    }
+    if (m_instanceId != id) {
+        m_instanceId = id;
+        initInstance();
+        emit instanceIdChanged();
+    }
+}
+
+void PluginEditorView::initInstance()
+{
+    if (m_activeViews.count(instanceId()) && m_activeViews[instanceId()]) {
+        auto activeWidget = m_activeViews[instanceId()];
+        m_instanceId = NOT_SETTED;
+        activeWidget->raise();
+        accept(); //set result code
+        return;
+    }
+    m_activeViews[instanceId()] = this;
+
+    auto instance = vstInstanceRegister()->instance(instanceId());
+    IF_ASSERT_FAILED(instance) {
+        return;
+    }
+    IF_ASSERT_FAILED(windowHandle()) {
+        LOGE() << "widget doen't have native window handle";
+        return;
+    }
+    m_view = instance->createView();
+
+    if (m_view) {
+        attachOriginalView();
+    } else {
+        attachQmlView(instance);
+    }
+}
+
+Steinberg::tresult PluginEditorView::resizeView(Steinberg::IPlugView* view, Steinberg::ViewRect* newSize)
+{
+    auto windowPosition = mainWindow()->qMainWindow()->pos();
+    QRect newGeometry(
+        windowPosition.x(),
+        windowPosition.y(),
+        newSize->getWidth(),
+        newSize->getHeight()
+        );
+    setGeometry(newGeometry);
+    view->onSize(newSize);
+    return kResultTrue;
+}
+
+void PluginEditorView::attachOriginalView()
+{
+#ifdef Q_OS_MAC
+    FIDString platformType = kPlatformTypeNSView;
+#elif Q_OS_IOS
+    FIDString platformType = kPlatformTypeUIView;
+#elif Q_OS_WIN
+    FIDString platformType = kPlatformTypeHWND;
+#elif Q_OS_LINUX
+    FIDString platformType = kPlatformTypeX11EmbedWindowID;
+#endif
+
+    if (m_view->isPlatformTypeSupported(platformType) != kResultTrue) {
+        LOGW() << "plugin doesn't support platform " << platformType;
+        return;
+    }
+
+    m_view->setFrame(this);
+
+    Steinberg::tresult attached;
+    attached = m_view->attached(reinterpret_cast<void*>(windowHandle()->winId()), platformType);
+    IF_ASSERT_FAILED_X(attached == kResultOk, "plugin's view was not attached") {
+        return;
+    }
+
+    // some plugins don't call resize after attached
+    ViewRect size;
+    m_view->getSize(&size);
+    resizeView(m_view, &size);
+}
+
+void PluginEditorView::attachQmlView(std::shared_ptr<PluginInstance> instance)
+{
+    setAttribute(Qt::WA_NativeWindow, false);
+    resize(640, 480);
+    setLayout(new QHBoxLayout);
+
+    auto widget = new QQuickWidget(QUrl("qrc:/qml/VSTEditor.qml"));
```

Comment:
There is a problem here, a new Qml engine will be created, and the global one in which everything works will not be used. Because of this, themes will not be available, and our modules will probably not be found ( UI components, VST itself). You need to make sure that the main Qml engine is used. You can see an example in `void QmlDialog::componentComplete()`

Reply:
fixed